lib/serialize-html.js

```diff
@@ -11,128 +30,77 @@
-                    if ((json.name && isWhite) || isSlash) {
-                        inTagName = false
-                        if (!json.name) {
-                            inCloser = true
-                            if (parent.parent) parent = parent.parent
-                        }
-                    } else if (isClose) {
-                        serialize(str.substr(i + 1), inClosing || inCloser ? parent : json)
-                        return parent
-                    } else if (!isWhite) {
-                        json.name += current
-                    }
-                // Attribute name
-                } else if (inAttrName || !attrName) {
-                    inAttrName = true
-                    if (attrName == null) attrName = ''
-                    if (isSlash ||
-                        (attrName && isWhite) ||
-                        (attrName && current == '=')) {
-
-                        inAttrName = false
-                        if (attrName) {
-                            if (!json.attributes) json.attributes = {}
-                            json.attributes[attrName] = ''
-                        }
-                    } else if (isClose) {
-                        serialize(str.substr(i + 1), inClosing || inCloser ? parent : json)
-                        return parent
-                    } else if (!isWhite) {
-                        attrName += current
-                    }
-                // Attribute value
-                } else if (inAttrValue || attrName) {
-                    if (attrValue == null) attrValue = ''
-
-                    if (isQuote) {
-                        if (inAttrValue) {
-                            if (current == openQuote) {
-                                if (attrValue) json.attributes[attrName] = attrValue
-                                inAttrValue = false
-                                attrName = attrValue = null
-                            } else {
-                                attrValue += current
-                            }
-                        } else {
-                            inAttrValue = true
-                            openQuote = current
-                        }
-                    } else if (inAttrValue) {
-                        attrValue += current
-                    }
-                }
-            } else if (isOpen) {
-                if (inText) {
-                    serialize(str.substr(i), parent)
-                    return parent
-                }
-                inTag = true
-            } else if (isSlash && !inAttrValue) {
-                end = true
-            } else {
-                inText = true
-                inTag = false
-                if (!json.name) json.name = '#text'
-                if (json.text == null) json.text = ''
-                json.text += current
-            }
+  var extractTag = (/^<(\/)?([a-z-]+)((?: [a-z:-]+="[^"]*")*)>$/)
+  var extractAttributes = (/[a-z]+="[^"]+"/g)
```

Comment:
That stuff is just sanity checking (check if we actually have a tag) and for parsing attributes